File: Session18/238. Product of Array Except Self.java
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n=len(nums)
        ans=[]
        
        rarr=[0]*n
        
        rarr[n-1]=1
        
        
        for i in range(n-2,-1,-1):
            rarr[i]=rarr[i+1]*nums[i+1]

        prev= 1
        ans.append(prev*rarr[0])
        
        for i in range(1,n):
            prev=prev*nums[i-1]
            ans.append(prev*rarr[i])
        
        return ans
        

File: Session18/238. Product of Array Except Self.py
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int n=nums.length;
        int larray[]=new int[n];
        int rarray[]=new int[n];
        larray[0]=1;
        for(int i=1;i<n;i++)
        {
            larray[i]=larray[i-1]*nums[i-1];
        }
        
        rarray[n-1]=1;
        for(int i=n-2;i>=0;i--)
        {
            rarray[i]=rarray[i+1]*nums[i+1];
        }
        
        for(int i=0;i<n;i++)
        {
            nums[i]=larray[i]*rarray[i];
        }
        
        return nums;
    }
}